### ISSUE TITLE
Gate socket timeout when refreshing health checks

### ISSUE THREAD

**Issue by jaredcurtis** [spinnaker/spinnaker#3442]
### Issue Summary:
Health checks are not being returned quickly enough before a timeout in Gate.

### Cloud Provider(s):
GCP, Kubernetes

### Environment:
Spinnaker Release 1.9.5
Clouddriver 3.4.5 (with and without https://github.com/spinnaker/clouddriver/pull/3015)
Deployed in Kubernetes
### Feature Area: 
Clouddriver
Gate

### Description:
When refreshing the cache or attempting to create/modify a server group the health check dropdown box is empty. Log messages are generated (see below) that indicate Gate is timing out before the results are returned.

This has been working previously. We suspect that the number of accounts we have in spinnaker is part of the problem. The reason for this suspicion is that we've recently added several new kube accounts which are in the APAC region and that when we noticed it stopped working.

```
hal config provider kubernetes account list | wc -l
Problems in halconfig:
- WARNING There is a newer version of Halyard available (1.11.0),
  please update when possible
? Run 'sudo apt-get update && sudo apt-get install
  spinnaker-halyard -y' to upgrade
100
```

### Additional Details:
```
time curl "http://spin-clouddriver.spinnaker:7002/search?q=&type=healthChecks&pageSize=1&page=1"
....
real    0m43.640s
user    0m0.009s
sys     0m0.003s
```

```
2018-10-10 22:37:53.551  INFO 1 --- [rix-searches-10] c.n.s.g.s.internal.ClouddriverService    : java.net.SocketTimeoutException: timeout
        at okio.Okio$4.newTimeoutException(Okio.java:232)
        at okio.AsyncTimeout.exit(AsyncTimeout.java:285)
        at okio.AsyncTimeout$2.read(AsyncTimeout.java:241)
        at okio.RealBufferedSource.indexOf(RealBufferedSource.java:355)
        at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:227)
        at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:221)
        at com.squareup.okhttp.internal.http.Http1xStream.readResponse(Http1xStream.java:186)
        at com.squareup.okhttp.internal.http.Http1xStream.readResponseHeaders(Http1xStream.java:127)
        at com.squareup.okhttp.internal.http.HttpEngine.readNetworkResponse(HttpEngine.java:739)
        at com.squareup.okhttp.internal.http.HttpEngine.access$200(HttpEngine.java:87)
        at com.squareup.okhttp.internal.http.HttpEngine$NetworkInterceptorChain.proceed(HttpEngine.java:724)
        at com.squareup.okhttp.internal.http.HttpEngine.readResponse(HttpEngine.java:578)
        at com.squareup.okhttp.Call.getResponse(Call.java:287)
        at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:243)
        at com.netflix.spinnaker.okhttp.OkHttpMetricsInterceptor.intercept(OkHttpMetricsInterceptor.java:44)
        at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:232)
        at com.squareup.okhttp.Call.getResponseWithInterceptorChain(Call.java:205)
        at com.squareup.okhttp.Call.execute(Call.java:80)                                                                                                     at retrofit.client.OkClient.execute(OkClient.java:53)
        at com.netflix.spinnaker.gate.retrofit.EurekaOkClient.super$2$execute(EurekaOkClient.groovy)                                                          at sun.reflect.GeneratedMethodAccessor158.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)                                                                           at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1213)
        at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuperN(ScriptBytecodeAdapter.java:132)
        at com.netflix.spinnaker.gate.retrofit.EurekaOkClient.execute(EurekaOkClient.groovy:82)
        at retrofit.RestAdapter$RestHandler.invokeRequest(RestAdapter.java:326)
        at retrofit.RestAdapter$RestHandler.invoke(RestAdapter.java:240)
        at com.sun.proxy.$Proxy108.search(Unknown Source)
        at com.netflix.spinnaker.gate.services.SearchService$_search_closure1.doCall(SearchService.groovy:36)
        at com.netflix.spinnaker.gate.services.SearchService$_search_closure1.call(SearchService.groovy)
        at com.netflix.spinnaker.security.AuthenticatedRequest.lambda$propagate$1(AuthenticatedRequest.java:79)
        at com.netflix.spinnaker.gate.services.commands.AbstractHystrixCommand.run(AbstractHystrixCommand.groovy:57)
        at com.netflix.hystrix.HystrixCommand$1.call(HystrixCommand.java:294)
        at com.netflix.hystrix.HystrixCommand$1.call(HystrixCommand.java:289)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
        at rx.Observable.unsafeSubscribe(Observable.java:10327)
        at rx.internal.operators.OnSubscribeDoOnEach.call(OnSubscribeDoOnEach.java:41)
        at rx.internal.operators.OnSubscribeDoOnEach.call(OnSubscribeDoOnEach.java:30)
        at rx.Observable.unsafeSubscribe(Observable.java:10327)
        at com.netflix.hystrix.AbstractCommand$5.call(AbstractCommand.java:520)
        at com.netflix.hystrix.AbstractCommand$5.call(AbstractCommand.java:495)
        at rx.Observable.unsafeSubscribe(Observable.java:10327)
        at rx.internal.operators.OperatorSubscribeOn$SubscribeOnSubscriber.call(OperatorSubscribeOn.java:100)
        at com.netflix.hystrix.strategy.concurrency.HystrixContexSchedulerAction$1.call(HystrixContexSchedulerAction.java:56)
        at com.netflix.hystrix.strategy.concurrency.HystrixContexSchedulerAction$1.call(HystrixContexSchedulerAction.java:47)
        at com.netflix.hystrix.strategy.concurrency.HystrixContexSchedulerAction.call(HystrixContexSchedulerAction.java:69)
        at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.net.SocketException: Socket closed
        at java.net.SocketInputStream.read(SocketInputStream.java:204)
        at java.net.SocketInputStream.read(SocketInputStream.java:141)
        at okio.Okio$2.read(Okio.java:140)
        at okio.AsyncTimeout$2.read(AsyncTimeout.java:237)
        ... 55 more

2018-10-10 22:37:53.552  INFO 1 --- [rix-searches-10] c.n.s.g.s.internal.ClouddriverService    : ---- END ERROR
2018-10-10 22:37:53.583 ERROR 1 --- [.0-8084-exec-40] c.n.s.g.s.c.AbstractHystrixCommand       : Downstream exception encountered (group: 'searches',
command: 'search-healthChecks')
```

**Comment by jaredcurtis** [comment]
We've upgraded our spinnaker release to 1.10.1 and can confirm this same issue is present.

**Comment by jaredcurtis** [comment]
Anyone have any suggestions on how I can attempt to resolve this?

**Comment by spinnakerbot** [comment]
This issue hasn't been updated in 45 days, so we are tagging it as 'stale'. If you want to remove this label, comment:

> @spinnakerbot remove-label stale

**Comment by jaredcurtis** [comment]
@spinnakerbot remove-label stale

What can I do to help move this issue forward?

**Comment by spinnakerbot** [comment]
This issue hasn't been updated in 45 days, so we are tagging it as 'stale'. If you want to remove this label, comment:

> @spinnakerbot remove-label stale

**Comment by spinnakerbot** [comment]
This issue is tagged as 'stale' and hasn't been updated  in 45 days, so we are tagging it as 'to-be-closed'. It will be closed in 45 days unless updates are made. If you want to remove this label, comment:

> @spinnakerbot remove-label to-be-closed

**Comment by expz** [comment]
I had this same error. It might not be pertinent to your case, but I put it here for anyone googling the error. I am not sure why the clouddriver was not responding, but it was emitting errors that  `the server doesn't have a resource type "podPreset"` as well as connection errors. It might have been caused by an installation that did not go smoothly (some IPs were blocked that I had to unblock, etc). What fixed it was
```
kubectl exec -it cd-spinnaker-halyard-0 /bin/bash
hal config provider kubernetes account edit spin --omit-kinds=[podPreset]
hal deploy apply
```
where `spin` is the context I used for deployment and `cd-spinnaker-halyard-0` is the name of my halyard pod. This redeployed everything (!), but once it was finished, the clouddriver errors were gone, and most health checks succeeded.
